ospfd/ospf_packet.c

```diff
@@ -2947,225 +2942,250 @@ int ospf_read(struct thread *thread)
 	uint16_t length;
 	struct interface *ifp = NULL;
 	struct connected *c;
+	int32_t count = 0;
 
 	/* first of all get interface pointer. */
 	ospf = THREAD_ARG(thread);
 
 	/* prepare for next packet. */
-	ospf->t_read = NULL;
 	thread_add_read(master, ospf_read, ospf, ospf->fd, &ospf->t_read);
 
-	stream_reset(ospf->ibuf);
-	ibuf = ospf_recv_packet(ospf, ospf->fd, &ifp, ospf->ibuf);
-	if (ibuf == NULL)
-		return -1;
-	/* This raw packet is known to be at least as big as its IP header. */
+	while (count < ospf->write_oi_count) {
```

Comment:
Count does not seem to be incremented it might end up reading all the packets on socket then?

Reply:
yes you are correct we would just read all available packets.  I have fixed this issue

Reply:
Code looks good, but I'd suggest to use `goto` or move this code into a function. The extra indentation makes the lines wrap more and causes code modification which we could avoid.

Reply:
Fixed @rzalamena